jwst/resample/resample.py

```diff
@@ -216,6 +215,57 @@ def blend_output_metadata(self, output_model):
             ignore=ignore_list
         )
 
+    def _get_intensity_scale(self, img):
+        """
+        Compute an intensity scale from the input and output pixel area.
+
+        For imaging data, the scaling is used to account for differences
+        between the nominal pixel area and the average pixel area for
+        the input data.
+
+        For spectral data, the scaling is used to account for flux
+        conservation with non-unity pixel scale ratios, when the
+        data units are flux density.
+
+        Parameters
+        ----------
+        img : DataModel
+            The input data model.
+
+        Returns
+        -------
+        iscale : float
+            The scale to apply to the input data before drizzling.
+        """
+        input_pixflux_area = img.meta.photometry.pixelarea_steradians
+        if input_pixflux_area:
+            if 'SPECTRAL' in img.meta.wcs.output_frame.axes_type:
+                # Use the nominal area as is
+                input_pixel_area = input_pixflux_area
+
+                # If input image is in flux density units, correct the
+                # flux for the user-specified change to the spatial dimension
+                if 'sr' not in str(img.meta.bunit_data).lower():
```

Comment:
I was wondering that too.  I looked for other examples in the jwst code for how we determine if the data is surface brightness or flux density, but didn't find anything useful for this situation.  I can see if astropy units might help.

Reply:
99.99% this will not work, but I was thinking along this line:

```python
is_equivalent(u.J, img.meta.bunit_data)
```
(replace `u.J` with whatever you expect data to be)

Reply:
Thanks, is_equivalent worked perfectly!  I added a `is_flux_density` helper function to resample_utils and called it here.